pkg/providers/amifamily/ami.go

```diff
@@ -246,13 +246,17 @@ func GetFilterAndOwnerSets(terms []v1beta1.AMISelectorTerm) (res []FiltersAndOwn
 			idFilter.Values = append(idFilter.Values, aws.String(term.ID))
 		default:
 			elem := FiltersAndOwners{
-				Owners: lo.Ternary(term.Owner != "", []string{term.Owner}, []string{"self", "amazon"}),
+				Owners: lo.Ternary(term.Owner != "", []string{term.Owner}, []string{}),
 			}
 			if term.Name != "" {
+				elem = FiltersAndOwners{
+					Owners: lo.Ternary(term.Owner != "", []string{term.Owner}, []string{"self", "amazon"}),
+				}
```

Comment:
Since this is only added in for name now, can you add a comment here so that future users know not to remove this because of a security issue?